lib/ansible/modules/database/postgresql/postgresql_sequence.py

```diff
@@ -0,0 +1,646 @@
+            ['absent', 'data_type'],
+            ['absent', 'increment'],
+            ['absent', 'minvalue'],
+            ['absent', 'maxvalue'],
+            ['absent', 'start'],
+            ['absent', 'cache'],
+            ['absent', 'cycle'],
+            ['absent', 'rename_to'],
+            ['absent', 'owner'],
+            ['absent', 'newschema'],
+            ['rename_to', 'data_type'],
+            ['rename_to', 'increment'],
+            ['rename_to', 'minvalue'],
+            ['rename_to', 'maxvalue'],
+            ['rename_to', 'start'],
+            ['rename_to', 'cache'],
+            ['rename_to', 'cycle'],
+            ['rename_to', 'cascade'],
+            ['rename_to', 'owner'],
+            ['rename_to', 'newschema'],
+            ['cascade', 'data_type'],
+            ['cascade', 'increment'],
+            ['cascade', 'minvalue'],
+            ['cascade', 'maxvalue'],
+            ['cascade', 'start'],
+            ['cascade', 'cache'],
+            ['cascade', 'cycle'],
+            ['cascade', 'owner'],
+            ['cascade', 'newschema'],
+        ]
+    )
+
+    # Note: we don't need to check mutually exclusive params here, because they are
+    # checked automatically by AnsibleModule (mutually_exclusive=[] list above).
+
+    # sequence = module.params["sequence"]
+    # state = module.params["state"]
+    # data_type = module.params["data_type"]
+    # increment = module.params["increment"]
+    # minvalue = module.params["minvalue"]
+    # maxvalue = module.params["maxvalue"]
+    # start = module.params["start"]
+    # cache = module.params["cache"]
+    # cycle = module.params["cycle"]
+    # cascade = module.params["cascade"]
+    # rename_to = module.params["rename_to"]
+    # schema = module.params["schema"]
+    # owner = module.params["owner"]
+    # newschema = module.params["newschema"]
+
+    # Connect to DB and make cursor object:
+    db_connection = connect_to_db(module, autocommit=True)
+    # Change autocommit to False if check_mode:
+    if module.check_mode:
+        db_connection.set_session(autocommit=False)
+    cursor = db_connection.cursor(cursor_factory=DictCursor)
+
+    ##############
+    # Create the object and do main job:
+    data = Sequence(module, cursor)
+
+    # Set defaults:
+    changed = False
+
+    # Create new sequence
+    if not data.exists and module.params['state'] == 'present':
+        if module.params.get('rename_to'):
+            module.fail_json(msg="Sequence '%s' does not exist, nothing to rename" % module.params['sequence'])
+        if module.params.get('newschema'):
+            module.fail_json(msg="Sequence '%s' does not exist, change of schema not possible" % module.params['sequence'])
+
+        changed = data.create()
+
+    # Drop non-existing sequence
+    elif not data.exists and module.params['state'] == 'absent':
+        # Nothing to do
+        module.warn(warning="Tries to drop nonexistent sequence '%s'" % module.params['sequence'])
```

Comment:
This should not result in a warning, but simply result in `changed == False`.

Reply:
The result is `changed == False` but I like to inform the user about why it does nothing. Is this not desirable, I can remove it.

Reply:
I would expect a module to issue warnings only for something which the user needs to be warned about. If the user wants something to be absent and it already is absent, that's not something I would expect a warning about. You also don't issue a warning if the user has `state == 'present'` and the sequence is already there.